### PR TITLE
move program status into a directive

### PR DESCRIPTION
**Pull request by elfenheart**

### CODE REVIEW

**Comment by anwar6953** [comment]
Any idea why Service program type is not applied here?

**Comment by elfenheart** [comment]
Service program type status container is a little different. I didn't want to change it just yet before I confirm

**Comment by anwar6953** [comment]
Ok, my understanding is that it's status should be inferred in the same way as other programs (like flow).

**Comment by elfenheart** [comment]
Right now services have instance container as the status. This will actually have to go through redesign first. We need to be able to increase the instance for the service as well. Will need to go to a different PR

**Comment by anwar6953** [comment]
Ok, so probably - the idea should be that instance container does not contain status, right?
And that the status should be displayed through the same way as other programs.
Maybe a different PR, but that should definitely be done, if agreed upon.

**Comment by elfenheart** [comment]
Yes, it should be consistent with CDAP services, flows, and services program.

File: cdap-ui/app/directives/program-status/program-status.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    return {
+      restrict: 'E',
+      scope: {
+        type: '@',
+        runid: '@'
+      },
+      templateUrl: 'program-status/program-status.html',
+      controller: function($scope, MyDataSource, $state) {
+        if ($scope.runid !== 'No Runs!') {
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/program-status/program-status.js]: Can we use $scope.runs.length property? At some point in time if we want to change the label for No Runs! to something else then this reference needs to be changed.

**Comment by ajainarayanan** [comment]
One Minor comment. Otherwise LGTM :+1:

**Comment by anwar6953** [comment]
@elfenheart https://github.com/caskdata/cdap/pull/2672#issuecomment-101817200
Can you then make it a point to do that in a follow up PR, then?

**Comment by elfenheart** [comment]
@anwar6953 Will be done with services instance

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**